logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStreamImpl.java

```diff
@@ -184,9 +186,14 @@ public void close() {
   }
 
   private LogStreamReader createLogStreamReader() {
-    final LogStreamReader newReader = new LogStreamReaderImpl(logStorage.newReader());
-    readers.add(newReader);
-    return newReader;
+    try {
+      final LogStreamReader newReader = new LogStreamReaderImpl(logStorage.newReader());
+      readers.add(newReader);
+      return newReader;
+    } catch (final UnrecoverableException e) {
```

Comment:
Instead of handling this case by case, you can handle this in `actor::handleFailure` method.